### PR TITLE
Migrate to magex

### PR DESCRIPTION
**Pull request by Cadair**
more magex less pfsspy

I also made it so that the tests will run without it.

### CODE REVIEW

File: sunkit_pyvista/plotter.py [hunk 1]
```diff
@@ -417,19 +417,19 @@ def plot_quadrangle(
 
     def plot_field_lines(self, field_lines, *, color_func=None, **kwargs):
         """
-        Plot magnetic field lines from `pfsspy`.
+        Plot magnetic field lines.
 
         Parameters
         ----------
-        field_lines : `pfsspy.fieldline.FieldLines`
+        field_lines : `sunkit_magex.pfss.fieldline.FieldLines`
```
Comment (nabobalis) [inline sunkit_pyvista/plotter.py:424]: Does this not need to be added to the docs config?

We should also enable nitpicky

Comment (Cadair) [reply sunkit_pyvista/plotter.py:424]: yes, I need to add it to intersphinx. I am not doing template or infra in this PR. I am on a fast path to working lol

File: sunkit_pyvista/tests/test_magex.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+import pyvista as pv
+import sunpy.map as smap
+from astropy.coordinates import SkyCoord
+from matplotlib import colors
+
+from sunkit_pyvista import SunpyPlotter
+
+pfss = pytest.importorskip("sunkit_magex.pfss")
+from sunkit_magex.pfss import tracing  # noqa: E402
+from sunkit_magex.pfss.sample_data import get_gong_map  # noqa: E402
```
Comment (nabobalis) [inline sunkit_pyvista/tests/test_magex.py:15]:
```suggestion
from sunkit_magex.pfss import tracing
from sunkit_magex.pfss.sample_data import get_gong_map
```

File: tox.ini [hunk 3]
```diff
@@ -28,7 +28,7 @@ extras =
     all
     docs
 commands =
-    sphinx-build -j auto --color -W --keep-going -b html -d _build/.doctrees . _build/html {posargs}
+    sphinx-build --color -W --keep-going -b html -d _build/.doctrees . _build/html {posargs}
```
Comment (nabobalis) [inline tox.ini:31]:
```suggestion
    sphinx-build -j auto --color -W --keep-going -b html -d _build/.doctrees . _build/html {posargs}
```
pydata fixed it but its not out, we can fail on this for now

**Comment by nabobalis** [comment]
Seems like windows just being windows.

**Review by nabobalis — APPROVED** [review]

File: sunkit_pyvista/tests/test_magex.py (hunk 2 above)
Comment (Cadair) [reply sunkit_pyvista/tests/test_magex.py:15]: Nope, these need to be below the pytest import or skip or it wont work.

File: tox.ini (hunk 3 above)
Comment (Cadair) [reply tox.ini:31]: I don't think it makes much difference here? it's not a big doc build project? I guess it wont break RTD tho.

**Comment by nabobalis** [comment]
Just add a changelog about the python bump and the magex change and i can merge it.

File: sunkit_pyvista/plotter.py (hunk 1 above)
Comment (Cadair) [reply sunkit_pyvista/plotter.py:424]: I can't seem to make intersphinx work

File: tox.ini (hunk 3 above)
Comment (nabobalis) [reply tox.ini:31]: Leave it for now then.

File: sunkit_pyvista/plotter.py (hunk 1 above)
Comment (nabobalis) [reply sunkit_pyvista/plotter.py:424]: I can do it after this PR.

**Pull request merged**